packages/modules/devices/solis/counter.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python3
+from dataclass_utils import dataclass_from_dict
+from modules.common.component_state import CounterState
+from modules.common.component_type import ComponentDescriptor
+from modules.common.fault_state import ComponentInfo, FaultState
+from modules.common.modbus import ModbusDataType, ModbusTcpClient_
+from modules.common.store import get_counter_value_store
+from modules.devices.solis.config import SolisCounterSetup
+from modules.devices.solis.version import SolisVersion
+
+
+class SolisCounter:
+    def __init__(self, component_config: SolisCounterSetup,
+                 version: SolisVersion) -> None:
+        self.component_config = dataclass_from_dict(SolisCounterSetup, component_config)
+        self.store = get_counter_value_store(self.component_config.id)
+        self.fault_state = FaultState(ComponentInfo.from_component_config(self.component_config))
+        self.version = version
+
+    def update(self, client: ModbusTcpClient_):
+        unit = self.component_config.configuration.modbus_id
+        if self.version == SolisVersion.inverter:
+            # register offset -1
+            power = client.read_holding_registers(3262, ModbusDataType.INT_32, unit=unit)
+            powers = client.read_holding_registers(3256, [ModbusDataType.INT_32]*3, unit=unit)
+            frequency = client.read_holding_registers(3281, ModbusDataType.UINT_16, unit=unit) / 100
+            imported = client.read_holding_registers(3282, ModbusDataType.UINT_32, unit=unit) * 10
+            exported = client.read_holding_registers(3284, ModbusDataType.UINT_32, unit=unit) * 10
+        elif self.version == SolisVersion.hybrid:
+            power = client.read_holding_registers(33263, ModbusDataType.INT_32, unit=unit)
+            powers = client.read_holding_registers(33257, [ModbusDataType.INT_32]*3, unit=unit)
+            frequency = client.read_holding_registers(33282, ModbusDataType.UINT_16, unit=unit) / 100
+            imported = client.read_holding_registers(33283, ModbusDataType.UINT_32, unit=unit) * 10
+            exported = client.read_holding_registers(33285, ModbusDataType.UINT_32, unit=unit) * 10
```

Comment:
Wenn Du das offset in eine Variable packst, wird es übersichtlicher:
```suggestion
        if self.version == SolisVersion.inverter:
            register_offset = 0
        elif self.version == SolisVersion.hybrid:
            register_offset = 1
        power = client.read_holding_registers(3262 + register_offset, ModbusDataType.INT_32, unit=unit)
        powers = client.read_holding_registers(3256 + register_offset, [ModbusDataType.INT_32]*3, unit=unit)
        frequency = client.read_holding_registers(3281 + register_offset, ModbusDataType.UINT_16, unit=unit) / 100
        imported = client.read_holding_registers(3282 + register_offset, ModbusDataType.UINT_32, unit=unit) * 10
        exported = client.read_holding_registers(3284 + register_offset, ModbusDataType.UINT_32, unit=unit) * 10
```